tests/Integration/Behaviour/Features/Context/CartRuleFeatureContext.php

```diff
@@ -0,0 +1,304 @@
+    }
+
+    /**
+     * @Given /^There is a cart rule with name (.+) and amount discount of (\d+) and priority of (\d+) and quantity of (\d+) and quantity per user of (\d+)$/
+     */
+    public function thereIsACartRuleWithNameAndAmountDiscountOfAndPriorityOfAndQuantityOfAndQuantityPerUserOf($cartRuleName, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser)
+    {
+        $this->createCartRule($cartRuleName, 0, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser);
+    }
+
+    protected function createCartRule($cartRuleName, $percent, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser)
+    {
+        $cartRule = new CartRule();
+        $cartRule->reduction_percent = $percent;
+        $cartRule->reduction_amount = $amount;
+        $cartRule->name = [Configuration::get('PS_LANG_DEFAULT') => $cartRuleName];
+        $cartRule->priority = $priority;
+        $cartRule->quantity = $cartRuleQuantity;
+        $cartRule->quantity_per_user = $cartRuleQuantityPerUser;
+        $now = new DateTime();
+        // sub 1s to avoid bad comparisons with strictly greater than
+        $now->sub(new DateInterval('P2D'));
+        $cartRule->date_from = $now->format('Y-m-d H:i:s');
+        $now->add(new DateInterval('P1Y'));
+        $cartRule->date_to = $now->format('Y-m-d H:i:s');
+        $cartRule->active = 1;
+        $cartRule->add();
+        $this->cartRules[$cartRuleName] = $cartRule;
+    }
+
+    /**
+     * @Given /^Cart rule named (.+) has a code: (.+)$/
+     */
+    public function cartRuleNamedHasACode($cartRuleName, $cartRuleCode)
+    {
+        $this->checkCartRuleWithNameExists($cartRuleName);
+        $this->cartRules[$cartRuleName]->code = $cartRuleCode;
+        $this->cartRules[$cartRuleName]->save();
+    }
+
+    /**
+     * @Given /^Cart rule named (.+) is restricted to product named (.+)$/
+     */
+    public function cartRuleNamedIsRestrictedToProductNamed($cartRuleName, $productName)
+    {
+        $this->checkCartRuleWithNameExists($cartRuleName);
+        if (!$this->productFeatureContext->productWithNameExists($productName)) {
+            throw new \Exception('Product with name "' . $productName . '" was not added in fixtures');
+        }
+        $this->cartRules[$cartRuleName]->product_restriction = true;
+        $this->cartRules[$cartRuleName]->reduction_product = $this->productFeatureContext->getProductWithName($productName)->id;
+        $this->cartRules[$cartRuleName]->save();
+    }
+
+    /**
+     * @Given /^Cart rule named (.+) is restricted to carrier named (.+)$/
+     */
+    public function cartRuleNamedIsRestrictedToCarrierNamed($cartRuleName, $carrierName)
+    {
+        $this->checkCartRuleWithNameExists($cartRuleName);
+        $this->carrierFeatureContext->checkCarrierWithNameExists($carrierName);
+        $this->cartRules[$cartRuleName]->carrier_restriction = 1;
+        $this->cartRules[$cartRuleName]->save();
+        Db::getInstance()->execute('
```

Comment:
I'm surprise there's nothing in CartRule ObjectModel to do that. Maybe it would be a relevant method to add ?